lib/widgets/action_sheet.dart

```diff
@@ -183,27 +178,39 @@ class MessageActionSheetCancelButton extends StatelessWidget {
   }
 }
 
-// This button is very temporary, to complete #125 before we have a way to
-// choose an arbitrary reaction (#388). So, skipping i18n.
-class AddThumbsUpButton extends MessageActionSheetMenuItemButton {
-  AddThumbsUpButton({super.key, required super.message, required super.pageContext});
+class ReactionButtons extends StatelessWidget {
+  const ReactionButtons({
+    super.key,
+    required this.message,
+    required this.pageContext,
+    required this.popularEmojis,
+  });
 
-  @override IconData get icon => ZulipIcons.smile;
+  final Message message;
 
-  @override
-  String label(ZulipLocalizations zulipLocalizations) {
-    return 'React with 👍'; // TODO(i18n) skip translation for now
-  }
+  /// A context within the [MessageListPage] this action sheet was
+  /// triggered from.
+  final BuildContext pageContext;
 
-  @override void onPressed() async {
+  /// List of popular emoji reaction buttons to display.
+  /// Each emoji must be a unicode emoji.
+  final List<EmojiCandidate> popularEmojis;
+
+  void _onPressed(
```

Comment:
Let's abstract this out into a function that both this and the emoji picker can call. Probably `emoji_reaction.dart` is a good home.